pkg/arvo/sys/vane/behn.hoon

```diff
@@ -202,33 +196,70 @@
     =*  timers     timers.state
     ::  if no timers, cancel existing wakeup timer or no-op
     ::
-    =/  timer=(unit [timer ~])  (peek:timer-map timers.state)
-    ?~  timer
+    =/  first=(unit [date=@da *])  (peek:timer-map timers.state)
+    ?~  first
       ?~  next-wake
         event-core
       (emit-doze ~)
     ::  if :next-wake is in the past or not soon enough, reset it
     ::
     ?^  next-wake
-      ?:  &((gte date.u.timer u.next-wake) (lte now u.next-wake))
+      ?:  &((gte date.u.first u.next-wake) (lte now u.next-wake))
         event-core
-      (emit-doze `date.u.timer)
+      (emit-doze `date.u.first)
     ::  there was no unix wakeup timer; set one
     ::
-    (emit-doze `date.u.timer)
-  ::  +set-timer: set a timer, maintaining the sort order of the :timers list
+    (emit-doze `date.u.first)
+  ::  +pop-timer: dequeue and produce earliest timer
+  ::
+  ++  pop-timer
+    ^+  [*timer timers.state]
+    =^  [date=@da dux=(qeu ^duct)]  timers.state  (pop:timer-map timers.state)
+    =^  dut  dux  ~(get to dux)
+    :-  [date dut]
+    ?:  =(~ dux)
+      +:(del:timer-map timers.state date)
```

Comment:
This is unnecessary and could be just `timers.state` because we updated it with `+pop`, right?

Reply:
Oh, you're right.